src/py42/services/cases.py

```diff
@@ -0,0 +1,194 @@
+        **kwargs
+    ):
+        """Gets an individual page of cases.
+
+        Args:
+            name (str, optional): Filter results by case name, matches partial names. Defaults to None.
+            status (str, optional): Filter results by case status, "OPEN" or "CLOSED". Defaults to None.
+            created_at (str, optional): Filter results by case creation time range, format ISO interval. Defaults to None. e.g 2020-08-31T11:00:00Z/2020-09-01T15:30:00Z
+            updated_at (str, optional): Filter results by last updated time range, format ISO interval. Defaults to None. e.g 2020-08-31T11:00:00Z/2020-09-01T15:30:00Z
+            subject (str, optional): Filter results based on User UID of a subject of a case. Defaults to None.
+            assignee (str, optional): Filter results based on User UID of an assignee of a case. Defaults to None.
+            page_number (int, optional): Page number of the results. Defaults to 1.
+            page_size (int, optional): Number of results to return per page. Defaults to 500.
+            sort_direction (str, optional): The direction on which to sort the response, based on the corresponding sort key. "asc" or "desc" Defaults to "asc".
+            sort_key (str, optional): Values on which the response will be sorted. Defaults to "number". Available options are
+             name, number, createdAt, updatedAt, status, assigneeUsername, subjectUsername
+
+        Returns:
+            :class:`py42.response.Py42Response`
+        """
+        param_dict = OrderedDict()
+        optional_fields = [
+            ("name", name),
+            ("subject", subject),
+            ("assignee", assignee),
+            ("createdAt", created_at),
+            ("updatedAt", updated_at),
+            ("status", status),
+        ]
+        for key, value in optional_fields:
+            if value is not None:
+                param_dict[key] = value
+
+        param_dict["pgNum"] = page_number
+        param_dict["pgSize"] = page_size
+        param_dict["srtDir"] = sort_direction
+        param_dict["srtKey"] = sort_key
+
+        params = "&".join(
+            [
+                "{}={}".format(key, value)
+                for key, value in zip(param_dict.keys(), param_dict.values())
+                if param_dict[key] is not None
+            ]
+        )
+        return self._connection.get("{}?{}".format(self._uri_prefix, params))
+
+    def get_all(
+        self,
+        name=None,
+        status=None,
+        created_at=None,
+        updated_at=None,
+        subject=None,
+        assignee=None,
+        page_number=1,
+        page_size=items_per_page,
+        sort_direction="asc",
+        sort_key="number",
+        **kwargs
+    ):
+        """Gets all cases.
+
+        Args:
+            name (str, optional): Filter results by case name, matches partial names. Defaults to None.
+            status (str, optional): Filter results by case status, "OPEN" or "CLOSED". Defaults to None.
```

Comment:
Probably will look better in the docs if we do ```OPEN``` and ```CLOSED```

Reply:
Changed.